### ISSUE TITLE
cpp_grpc_library fails to build on MacOS

### ISSUE THREAD

**Issue by mmende** [rules-proto-grpc/rules_proto_grpc#185]
### Issue Description

When trying to build one of our C++ GRPC/Protobuf projects yesterday without any major changes the build suddenly started to fail on MacOS. It seems like an error is being raised because in `com_google_absl/absl` an implicit copy constructor is defined which is deprecated and for this target the flags `-Werror,-Wdeprecated-copy` are set.  The build still runs without any issues on Ubuntu interestingly. I upgraded rules_proto_grpc from  `3.1.1` to `4.1.1` to try to fix the error but the same error appears. Upgrading bazel from `4.2.2` to `5.0.0` also didn't have any effect on the issue. To rule out the possibility that any other tool on my mac changed and caused this problem I let a coworker try to build the project with the same result. He also didn't change anything on his machine and could previously build the project without any issues. I also tried to add `-Wno-deprecated` to copts but `-Werror,-Wdeprecated-copy` seems to be prioritized unfortunately.

Is there any way of fixing the used grpc version to prevent such issues?

### Log Output

```plain text
INFO: Analyzed target //src:api_grpc (76 packages loaded, 4044 targets configured).
INFO: Found 1 target...
ERROR: /private/var/tmp/_bazel_mmende/18256cb928d1a5467132b2aa24b5aed1/external/upb/upbc/BUILD:44:10: Compiling upbc/protoc-gen-upb.cc [for host] failed: (Aborted): wrapped_clang_pp failed: error executing command external/local_config_cc/wrapped_clang_pp '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -g0 -O2 -DNDEBUG ... (remaining 58 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from external/upb/upbc/protoc-gen-upb.cc:28:
In file included from external/com_google_absl/absl/container/flat_hash_map.h:40:
In file included from external/com_google_absl/absl/container/internal/hash_function_defaults.h:56:
In file included from external/com_google_absl/absl/strings/cord.h:78:
external/com_google_absl/absl/functional/function_ref.h:124:16: error: definition of implicit copy constructor for 'FunctionRef<void (absl::string_view)>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
  FunctionRef& operator=(const FunctionRef& rhs) = delete;
               ^
external/com_google_absl/absl/strings/cord.h:1325:33: note: in implicit copy constructor for 'absl::FunctionRef<void (absl::string_view)>' first required here
    return ForEachChunkAux(rep, callback);
                                ^
1 error generated.
Error in child process '/usr/bin/xcrun'. 1
Target //src:api_grpc failed to build
Use --verbose_failures to see the command lines of failed build steps.
ERROR: /private/var/tmp/_bazel_mmende/18256cb928d1a5467132b2aa24b5aed1/external/com_github_grpc_grpc/src/proto/grpc/gcp/BUILD:19:14 Middleman _middlemen/@com_Ugithub_Ugrpc_Ugrpc_S_Ssrc_Sproto_Sgrpc_Sgcp_Calts_Uhandshaker_Uproto.upb-BazelCppSemantics_build_arch_darwin-fastbuild failed: (Aborted): wrapped_clang_pp failed: error executing command external/local_config_cc/wrapped_clang_pp '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -g0 -O2 -DNDEBUG ... (remaining 58 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox
INFO: Elapsed time: 84,857s, Critical Path: 19,53s
INFO: 494 processes: 52 internal, 442 darwin-sandbox.
FAILED: Build did NOT complete successfully
```

### rules_proto_grpc Version

3.1.1, 4.1.1

### Bazel Version

4.2.2, 5.0.0

### OS

macOS Monterey 12.2.1 (same issue with 12.3)

### WORKSPACE Content

```Starlark
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "rules_proto_grpc",
    sha256 = "507e38c8d95c7efa4f3b1c0595a8e8f139c885cb41a76cab7e20e4e67ae87731",
    #sha256 = "7954abbb6898830cd10ac9714fbcacf092299fda00ed2baf781172f545120419",
    #strip_prefix = "rules_proto_grpc-3.1.1",
    #urls = ["https://github.com/rules-proto-grpc/rules_proto_grpc/archive/3.1.1.tar.gz"],
    strip_prefix = "rules_proto_grpc-4.1.1",
    urls = ["https://github.com/rules-proto-grpc/rules_proto_grpc/archive/4.1.1.tar.gz"],
)

load("@rules_proto_grpc//:repositories.bzl", "rules_proto_grpc_repos", "rules_proto_grpc_toolchains")

rules_proto_grpc_toolchains()

rules_proto_grpc_repos()

load("@rules_proto//proto:repositories.bzl", "rules_proto_dependencies", "rules_proto_toolchains")

rules_proto_dependencies()

rules_proto_toolchains()

load("@rules_proto_grpc//cpp:repositories.bzl", rules_proto_grpc_cpp_repos = "cpp_repos")

rules_proto_grpc_cpp_repos()

load("@com_github_grpc_grpc//bazel:grpc_deps.bzl", "grpc_deps")

grpc_deps()
```

### BUILD Content

```Starlark
load("@rules_proto_grpc//cpp:defs.bzl", "cpp_grpc_compile", "cpp_grpc_library")

cpp_grpc_library(
    name = "api_grpc",
    protos = [
        "//protos:services_proto",
        "//protos:types_proto",
    ],
)
```

**Comment by aaliddell** [comment]
I don't have a Mac machine to test on, but do you have protobuf/protoc/absl installed by any other means on the machine? e.g. brew. And has that changed recently?

I've seen before that the sandboxing in macOS Bazel is less strict than on Linux, so am wondering if there is conflicting library versions on the machine?

**Comment by mmende** [comment]
Yes, I had the protobuf package installed over brew which apparently comes with the protoc binary. However I doubt that this is the problem because I can still compile the proto targets directly without any issues when running `bazel build //protos:services_proto` for example. There is also a `cpp_grpc_compile` target to create reference .cc and .h files which also compiles fine. The brew version of absl is not installed on my system. To verify that the brew protobuf package is not the problem I just uninstalled it temporarily and tried to build the `cpp_grpc_library` target again with the same problem unfortunately.

**Comment by mmende** [comment]
Awesome, installing XCode 13.2.1 additionally and switching the command line tools to 13.2.1 in `XCode -> Preferences -> Locations -> Command Line Tools` before starting the build did the trick. Thanks for your support.

**Comment by codingrhythm** [comment]
But we cannot stay on 13.2.1 forever, this error still happens to me. Will there be a fix for this soon?

**Comment by mmende** [comment]
Yeah true. But as this issue originated from [abseil/abseil-cpp](https://github.com/abseil/abseil-cpp) (and I could resolve it temporarily) I closed it. It may be a better idea to reopen the [mentioned issue](https://github.com/abseil/abseil-cpp/issues/1141) at abseil/abseil-cpp...
